packages/preferences/src/browser/preference-frontend-module.ts

```diff
@@ -6,17 +6,17 @@
  */
 
 import { ContainerModule, interfaces, } from 'inversify';
-import { PreferenceProviders } from "@theia/core/lib/browser/preferences";
+import { PreferenceProviders, PreferenceProvider, PreferenceScope } from "@theia/core/lib/browser/preferences";
 import { UserPreferenceProvider } from './user-preference-provider';
 import { WorkspacePreferenceProvider } from './workspace-preference-provider';
 
 export function bindPreferences(bind: interfaces.Bind, rebind: interfaces.Rebind): void {
-    bind(UserPreferenceProvider).toSelf().inSingletonScope();
-    bind(WorkspacePreferenceProvider).toSelf().inSingletonScope();
+    bind(PreferenceProvider).to(UserPreferenceProvider).inSingletonScope().whenTargetNamed(PreferenceScope.User);
```

Comment:
We should also bind stub implementations in the core. In the case if `preferences` extension is not installed it should not break.

You can convert `PreferenceProvider` from interface to such stub implementation.

Reply:
done